model_analyzer/entrypoint.py

```diff
@@ -0,0 +1,340 @@
+        if metric in [GPUFreeMemory, GPUUsedMemory]:
+            dcgm_tags.append(metric)
+        elif metric in [PerfThroughput, PerfLatency]:
+            perf_tags.append(metric)
+    return perf_tags, dcgm_tags
+
+
+def create_output_table(param_headers,
+                        perf_tags,
+                        dcgm_tags,
+                        aggregation_tag='Max'):
+    """
+    Creates and output table
+    with the correct headers
+
+    Parameters
+    ----------
+    param_headers : List of str
+        A list of names of parameters that will also
+        have columns in the table e.g. ['Model', 'Batch']
+    perf_tags : List of Record types
+        The metrics to be obtained from perf_analyzer
+    dcgm_tags : List of Record types
+        The metrics to be obtained from DCGMMonitor
+    aggregation_tag : str
+        Name that represents the aggregation function
+        used to aggregate the records
+
+    Returns
+    -------
+    OutputTable
+        A table configured with the required headers
+    """
+
+    table_headers = param_headers[:]
+    table_headers += [metric.header() for metric in perf_tags]
+    table_headers += [
+        aggregation_tag + ' ' + metric.header() for metric in dcgm_tags
+    ]
+    return OutputTable(headers=table_headers)
+
+
+def profile(args, perf_analyzer, dcgm_monitor, perf_tags, dcgm_tags):
+    """
+    Does a single run of perf_analyzer and 
+    collects metrics.
+
+    Parameters
+    ----------
+    args : namespace
+        The arguments passed into the CLI
+    perf_analyzer : PerfAnalyzer
+        instance used to get performance metrics
+    dcgm_monitor : DCGMMonitor
+        The DCGM monitoring instance 
+    perf_tags : List of Record types
+        The metrics to be obtained from perf_analyzer
+    dcgm_tags : List of Record types
+        The metrics to be obtained from DCGMMonitor
+
+    Returns
+    -------
+    dict
+        keys are metric types
+        values are their values
+    """
+
+    # Start monitors and run perf_analyzer
+    dcgm_monitor.start_recording_metrics()
+    time.sleep(0.1)
+    try:
+        perf_records = perf_analyzer.run(perf_tags)
+    except FileNotFoundError as e:
+        raise TritonModelAnalyzerException(
+            f"perf_analyzer binary not found : {e}")
+    dcgm_records = dcgm_monitor.stop_recording_metrics()
+
+    # Insert all records into aggregator and get aggregated DCGM records
+    record_aggregator = RecordAggregator()
+    for record in perf_records + dcgm_records:
+        record_aggregator.insert(record)
+    aggregated_records = record_aggregator.aggregate(record_types=(perf_tags +
+                                                                   dcgm_tags),
+                                                     reduce_func=max)
+    return aggregated_records
+
+
+def main():
+    """
+    Main entrypoint of model_analyzer
+    """
+
+    # Get args and set monitoring metrics
+    args = CLI().parse(sys.argv)
```

Comment:
You are still mixing levels of abstraction and main is still too long. Try to further extract methods so that main is less than 15 lines of code.  Off the top of my head, I can see something like....
```
def main():
   ... = initialize_model_analyzer(...)
   ... = start_triton(...)
   ... = profile_models(...)
   ... = stop_triton(...)
   ... = write_reports(...)
```

Note how in five lines of code, I clearly see the top level business logic of what ModelAnalyzer is doing. Everything is at the same high level of abstraction. And you don't need "section comments".  

Note that if you find yourself needing  a comment to explain what the next block of code does, then you should  extract a method.

Reply:
Also note that you create several methods above. Considering grouping methods together that share the same purpose and creating a class to contain them.  This simple refactoring is know as Extract Class (https://refactoring.guru/extract-class).